handlers/meta/whatsapp/templates.go

```diff
@@ -39,3 +44,67 @@ func GetTemplating(msg courier.MsgOut) (*MsgTemplating, error) {
 
 	return metadata.Templating, nil
 }
+
+func GetTemplatePayload(templating MsgTemplating, lang string) *Template {
+	template := Template{Name: templating.Template.Name, Language: &Language{Policy: "deterministic", Code: lang}}
+
+	for k, v := range templating.Params {
+		if strings.HasPrefix(k, "button.") {
+
+			for _, p := range v {
+				if strings.HasPrefix(p.Value, "http") {
+					component := &Component{Type: "button", Index: strings.TrimPrefix(k, "button."), SubType: "quick_reply"}
+					component.Params = append(component.Params, &Param{Type: "url", Text: p.Value})
+					template.Components = append(template.Components, component)
+				} else {
+					component := &Component{Type: "button", Index: strings.TrimPrefix(k, "button."), SubType: "quick_reply"}
+					component.Params = append(component.Params, &Param{Type: "payload", Payload: p.Value})
+					template.Components = append(template.Components, component)
+				}
+			}
+
+		} else if k == "header" {
+			component := &Component{Type: "header"}
+			for _, p := range v {
+				if p.Type == "image" {
+					component.Params = append(component.Params, &Param{Type: p.Type, Image: &struct {
+						Link string "json:\"link,omitempty\""
+					}{Link: p.Value}})
+				} else if p.Type == "video" {
+					component.Params = append(component.Params, &Param{Type: p.Type, Video: &struct {
+						Link string "json:\"link,omitempty\""
+					}{Link: p.Value}})
+				} else if p.Type == "document" {
+					component.Params = append(component.Params, &Param{Type: p.Type, Document: &struct {
+						Link string "json:\"link,omitempty\""
+					}{Link: p.Value}})
+				} else {
+					component.Params = append(component.Params, &Param{Type: p.Type, Text: p.Value})
+				}
+			}
+			template.Components = append(template.Components, component)
+
+		} else {
+			component := &Component{Type: "body"}
+			for _, p := range v {
+				component.Params = append(component.Params, &Param{Type: p.Type, Text: p.Value})
+			}
+			template.Components = append(template.Components, component)
+
+		}
+
+	}
+
+	if len(templating.Params) == 0 {
+		component := &Component{Type: "body"}
+
+		for _, v := range templating.Variables {
```

Comment:
I kept supporting the templating variables so this can be deployed even if the goflow changes are not yet done